php/class-customize-snapshot-manager.php

```diff
@@ -123,6 +123,11 @@ public function __construct( Plugin $plugin ) {
 		add_action( 'customize_save_after', array( $this, 'save_snapshot' ) );
 		add_action( 'admin_bar_menu', array( $this, 'customize_menu' ), 41 );
 		add_action( 'customize_controls_print_footer_scripts', array( $this, 'render_templates' ) );
+		
+		add_filter( 'theme_mod_nav_menu_locations', array( $this, 'filter_theme_mod_nav_menu_locations' ) );
+		add_filter( 'wp_get_nav_menus', array( $this, 'filter_wp_get_nav_menus' ) );
+		add_filter( 'wp_get_nav_menu_items', array( $this, 'filter_wp_get_nav_menu_items' ), 10, 3 );
+		add_filter( 'wp_get_nav_menu_object', array( $this, 'filter_wp_get_nav_menu_object' ), 10, 2 );
```

Comment:
Would we not be able to re-use the methods in `WP_Customize_Nav_Menus`? https://github.com/xwp/wordpress-develop/blob/57d2c46591e6593c8d6bfcb2100b6538e80f9dce/src/wp-includes/customize/class-wp-customize-nav-menu-setting.php#L231-L234

That is, use a similar approach to re-using methods from `WP_Customize_Widgets`?